src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,153 @@
+        }
+        return newTable;
+    }
+
+    private void passAcrossExtendedTableElemElementNodesToPutTransferredNode(int newTableIndex, Node<K,V>[] newTable, Node<K,V> transferredNode) {
+        Node<K,V> currentNodeOfTableToBeAddedTo = newTable[newTableIndex];
+        boolean areNodeKeysEqual = false;
+        while (currentNodeOfTableToBeAddedTo.next != null) {
+            if (transferredNode.key == currentNodeOfTableToBeAddedTo.key
+                    || transferredNode.key != null && transferredNode.key.equals(currentNodeOfTableToBeAddedTo.key)) {
+                currentNodeOfTableToBeAddedTo.value = transferredNode.value;
+                areNodeKeysEqual = true;
+                break;
+            } else {
+                currentNodeOfTableToBeAddedTo = currentNodeOfTableToBeAddedTo.next;
+            }
+        }
+        if (!areNodeKeysEqual) {
+            currentNodeOfTableToBeAddedTo.next = new Node<K, V>(transferredNode.key, transferredNode.value, null);
+        }
+    }
+
+    private boolean passAcrossCurrentCapacityTableElementNodesToPutNewNode(Node<K,V>[] tableToBeAddedTo, int index, Node<K,V> nodeToPut, boolean isSizeIncrease) {
+        Node<K,V> currentNodeOfTableToBeAddedTo = tableToBeAddedTo[index];
+        while (currentNodeOfTableToBeAddedTo != null) {
+            if (nodeToPut.key == currentNodeOfTableToBeAddedTo.key
+                    || nodeToPut.key != null && nodeToPut.key.equals(currentNodeOfTableToBeAddedTo.key)) {
+                currentNodeOfTableToBeAddedTo.value = nodeToPut.value;
+                isSizeIncrease = false;
+                break;
+            }
+            if (currentNodeOfTableToBeAddedTo.next == null) {
+                break;
+            }
+            currentNodeOfTableToBeAddedTo = currentNodeOfTableToBeAddedTo.next;
+        }
+        if (isSizeIncrease) {
+            currentNodeOfTableToBeAddedTo.next = new Node<K, V>(nodeToPut.key, nodeToPut.value, null);
+        }
+        return isSizeIncrease;
+    }
+
+    private void increaseCurrentCapacityAndThreshold() {
+        currentCapacity = currentCapacity << 1;
+        threshold = threshold << 1;
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        int indexOfSearchedElement = calculateIndex(key);
+        if (table[indexOfSearchedElement] == null) {
+            return null;
+        }
+        Node<K,V> currentNode = table[indexOfSearchedElement];
+        Node<K,V> foundElement = null;
+        while (currentNode != null) {
+            if (currentNode.key == key || key != null && key.equals(currentNode.key)) {
+                foundElement = currentNode;
+                break;
+            }
+            currentNode = currentNode.next;
+        }
+        return foundElement != null ? foundElement.value : null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private static class Node<K,V> {
+        private int hash;
```

Comment:
The `hash` field in the `Node` class is declared but never used. If it's not needed, consider removing it to clean up the code.